### ISSUE TITLE
Deleting lines inside code blocks leaves void space

### ISSUE THREAD

**Issue by nvartolomei** [ProseMirror/prosemirror#1066]
### Steps to reproduce

1. Enter code_block mode by typing \```
2. Write 3 lines of code
3. Go to second line and press backspace until you delete it completely and cursor jumps to the end of the first line
4. Observe that in the place of second line which we just tried to delete now is a void space

### ProseMirror version

Latest on npm, latest on the ProseMirror website.

### Affected platforms

- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer
- [x] Other (Safari Version 13.1.1 (15609.2.9.1.2))

### Screenshots / Screencast (Optional)

![Screen Recording 2020-06-10 at 23 08 04](https://user-images.githubusercontent.com/543193/84324306-68228100-ab70-11ea-93fa-7391d0bc9e6a.gif)

The most odd thing is that I can trigger a somewhat similar behavior in this GitHub issue editor textarea.

**Comment by marijnh** [comment]
I can only test with Safari 12 (due to Apple apparently tying Safari upgrades to OS upgrades), but there I don't see the issue. If this is also happening in other editors, maybe check if you have any browser extensions installed that cause this?

**Comment by nvartolomei** [comment]
@marijnh disabled all extensions, still reproducible. Re github issue editor textarea, it seems to be prosemirror actually but integrated in a peculiar way (let's ignore this fact).

**Comment by nvartolomei** [comment]
Minimal reproducible example: https://codesandbox.io/s/interesting-kare-0ls2i

Put caret on line 2, delete it, void space remains there. It's interesting that if you press enter inside this vanity contenteditable and then delete characters them it behaves as you would expect. Puzzled.

```html
<p
  id="test"
  style="white-space: pre-wrap; border: 1px solid"
  contenteditable="true"
/>
<script>
  document.getElementById("test").innerHTML = "line1\nline2\nline3";
</script>
```

**Comment by marijnh** [comment]
Right, I can reproduce this now. Seems Webkit is displaying the content in a way that just doesn't correspond to what the DOM structure holds (there's a text node with a single newline in it, yet it displays a blank line as if there are two). I don't know an obvious way for ProseMirror to work around that. I think your best bet is to report this as a Webkit bug and hope someone looks at it on that side.

**Comment by marijnh** [comment]
This issue has been moved to [issue #1066](https://code.haverbeke.berlin/prosemirror/prosemirror/issues/1066) on the [repository's new home](https://code.haverbeke.berlin/prosemirror/prosemirror/).
